fingerprint.c

```diff
@@ -44,14 +51,32 @@ typedef struct {
     char db_path[255];
     pthread_mutex_t lock;
     uint64_t challenge;
+    int state;
```

Comment:
name the enum something like `worker_state` and use `enum worker_state state;` here